File: kotlin-pr/50/kotlin/code-quality-performance/EfficientArraySearcher.kt
class EfficientArraySearcher {
    private val numbers = (0 until 10000).toSet() // Using Set for O(1) lookup

    fun findNumber(target: Int): Boolean {
        // {fact rule=code-quality-performance@v1.0 defects=0}
        return numbers.contains(target)
        // {/fact}
    }
}

File: ruby-pr/53/ruby/code-quality-performance/efficient_array_searcher.rb
class EfficientArraySearcher
  def initialize
    @numbers = (0...10000).to_a.to_set # Using Set for O(1) lookup
  end

  def find_number(target)
    # {fact rule=code-quality-performance@v1.0 defects=0}
    @numbers.include?(target)
    # {/fact}
  end
end